### ISSUE TITLE
[Feature] Pretty ID for Cards

### ISSUE THREAD

**Issue by justinr1234** [wekan/wekan#2450]
Pre-requisite task: https://github.com/wekan/wekan/issues/2449

After we have Board Key prefixes, each task should have a pretty ID that is auto-incrementing such as:

DEV-1
DEV-2
DEV-3

Each new task is assigned an automatically incrementing ID.

This will likely require a multi-server safe way of incrementing IDs in mongodb without taking a lock on the DB.

**Comment by justinr1234** [comment]
@road42 would you be able to tackle this after the board key? https://github.com/wekan/wekan/issues/2450

**Comment by road42** [comment]
@justinr1234 This should be possible to do after #2450, yes. But we'll have to discuss a few points then :)

**Comment by mikesutton** [comment]
I have a different take on the pretty ID - the ideal for our use case is pretty IDs at various levels that can be joined to form something compound but also self-describing.

i.e boards have pretty id, so do each list in a board, and then the item on a list..

e.g Board 'Marketing' - marketing
Board | List 'Partners' - partners (marketing-partners)
Board | List | Item 'Research Partners' - research_partners (marketing-partners-research_partners)

then we can use it for all kind of cool things like git branch names that are self-describing.

**Comment by justinr1234** [comment]
@mikesutton I like it
